test/tests/test_optional_argument.cpp

```diff
@@ -139,6 +139,62 @@ TEST_CASE_FIXTURE(
     REQUIRE_FALSE(sut_has_value(sut));
 }
 
+TEST_CASE_FIXTURE(
+    optional_argument_test_fixture,
+    "choices(any) should throw when passed choices are not of test_value_type"
```

Comment:
Should be `choices(vector<any>)`

If you remove the abstract method in the interface you could get rid of this test case